content/learning-paths/cross-platform/simd-info-demo/simdinfo-example1-porting.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Code Verification
+weight: 6
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+### Step-by-Step Porting
+
+1. Change the loading process to follow **NEON**'s method for initializing vectors. The **SSE4.2** intrinsic **`_mm_set_ps`** is in reality a macro, in **NEON** you can do the same thing with curly braces **`{}`** inititialization.
+2. Next, you will replace the **SSE4.2** intrinsics with the **NEON** equivalents we identified earlier. The key is to ensure that the operations perform the same tasks, such as comparison, addition, multiplication, and square root calculations.
+3. Finally, modify the storing process to match **NEON**’s way of moving data from vectors to memory. In **NEON**, you use functions like [**`vst1q_f32`**](https://simd.info/c_intrinsic/vst1q_f32/) for storing 128-bit floating-point vectors and [**`vst1q_u32`**](https://simd.info/c_intrinsic/vst1q_u32/) for storing 128-bit integer vectors.
+
+After identifying the **NEON** intrinsics you will need in the ported program, it's time to actually write the code.
+
+Create a new file for the ported NEON code named `calculation_neon.c` with the contents shown below:
+
+```C
+#include <arm_neon.h>
+#include <stdio.h>
+
+int main() {
+    float32x4_t a = {1.0f, 4.0f, 9.0f, 16.0f};
+    float32x4_t b = {1.0f, 2.0f, 3.0f, 4.0f};
```

Comment:
This is undefined behavior for ACLE and would put the values in the wrong order for Big-Endian.
Initialization should be done through a load
```c
    float32_t a_array[4] = {1.0f, 4.0f, 9.0f, 16.0f};
    float32_t b_array[4] = {1.0f, 2.0f, 3.0f, 4.0f};
    float32x4_t a = vld1q_f32 (a_array);
    float32x4_t b = vld1q_f32 (b_array);
```
which allows the compiler to do the lane correction on Big-Endian. Same for the other examples.

Reply:
Thank you for the feedback! We originally focused on little-endian, but we've now updated the code to ensure compatibility with big-endian systems.